crates/vizia_core/src/systems/text.rs

```diff
@@ -73,79 +72,91 @@ pub fn build_paragraph(
-                families
-                    .iter()
-                    .map(|family| match family {
-                        FamilyOwned::Generic(generic) => match generic {
-                            _ => "Roboto Flex",
-                        },
-
-                        FamilyOwned::Named(name) => name.as_str(),
-                    })
-                    .collect()
-            })
-            .unwrap_or(vec!["Roboto Flex"]);
-
-        text_style.set_font_families(&font_families);
-
-        text_style.set_color(style.font_color.get(entity).copied().unwrap_or_default());
-
-        let font_size = style.font_size.get(entity).copied().map(|f| f.0).unwrap_or(16.0)
-            * style.scale_factor();
-        text_style.set_font_size(font_size);
-        // text_style.add_font_feature("opsz", 1);
-        let coordinates = Box::new([
-            Coordinate { axis: ('w', 'g', 'h', 't').into(), value: font_weight.0 as f32 },
-            Coordinate { axis: ('o', 'p', 's', 'z').into(), value: font_size },
-        ]);
-        let args = FontArguments::new();
-        let pos = VariationPosition { coordinates: coordinates.as_ref() };
-        let args = args.set_variation_design_position(pos);
-        text_style.set_font_arguments(&Some(args));
-
-        paragraph_style.turn_hinting_off();
-        paragraph_style.set_text_style(&text_style);
-        let mut paragraph_builder = ParagraphBuilder::new(&paragraph_style, font_collection);
-
-        paragraph_builder.push_style(&text_style);
-        paragraph_builder.add_text(text.as_str());
-        paragraph_builder.add_text("\u{200B}");
-
-        paragraph_builder.build()
-    })
+
+        // Font Families
+        text_style.set_font_families(
+            style
+                .font_family
+                .get(entity)
+                .map(Vec::as_slice)
+                .unwrap_or(&[FamilyOwned::Generic(GenericFontFamily::SansSerif)]),
+        );
+
+        // Font Color
+        if let Some(font_color) = style.font_color.get(entity) {
+            let mut paint = Paint::default();
+            paint.set_color(*font_color);
+            paint.set_anti_alias(true);
+            text_style.set_foreground_paint(&paint);
+        }
+
+        // Font Size
+        let font_size = style.font_size.get(entity).map_or(16.0, |f| f.0);
+        text_style.set_font_size(font_size.round() * style.scale_factor());
+
+        // Font Style
+        match (
+            style.font_weight.get(entity),
+            style.font_width.get(entity),
+            style.font_slant.get(entity),
+        ) {
+            (None, None, None) => {}
+            (weight, width, slant) => {
+                text_style.set_font_style(FontStyle::new(
+                    weight.copied().unwrap_or_default().into(),
+                    width.copied().unwrap_or_default().into(),
+                    slant.copied().unwrap_or_default().into(),
+                ));
+            }
+        }
```

Comment:
Not sure I understand this part. Wouldn't this only set the font style if all of weight, width, and slant are defined?

Reply:
No, the idea was that it will not set a font style if *all* of them are `None`. If any are not `None` then it will take the 2nd match.

Partially as an optimization, but mainly I thought it would be best to avoid applying the font style unless its explicitly requested. To try and not conflict with variation parameters. 

Though now that I look at the MSDN page, it seems that the correct thing to do is have it so variations always [override the equivalent font style](https://developer.mozilla.org/en-US/docs/Web/CSS/font-variation-settings#description). Though I'm not sure what the unit conversions should be in that case (like at what threshold of `slnt` do you become oblique/italic?).

Reply:
Ah yes of course. Do we know whether font arguments override the equivalent font style in skia? I thought they did but I can't remember

Reply:
I just tested and skia does not handle this for us unfortunately.
```cs
    font-weight: bold;
    font-variation-settings: 'wght' 320;
```
These combined together look different than either one does in isolation.